SDK/AppCenterData/Microsoft.AppCenter.Data.Shared/ConversionExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+namespace Microsoft.AppCenter.Data
+{
+    public static class SharedConversionExtensions
+    {
+        public static DocumentWrapper<T> ToDocumentWrapper<T>(
+            string partition,
+            string eTag,
+            string id,
+            bool isFromDeviceCache,
+            string jsonValue,
+            Func<string, T> deserealizer,
+            DateTimeOffset date)
+        {
+            DocumentWrapper<T> parsedDocument = new DocumentWrapper<T>
+            {
+                Partition = partition,
+                ETag = eTag,
+                Id = id,
+                IsFromDeviceCache = isFromDeviceCache,
+                JsonValue = jsonValue,
+                DeserializedValue = default(T),
+                LastUpdatedDate = date
+            };
+
+            // Try to deserialize.
+            try
+            {
+                // Set deserialized value.
+                if (parsedDocument.JsonValue != null)
+                {
+                    parsedDocument.DeserializedValue = deserealizer(parsedDocument.JsonValue);
+                }
+            }
+            catch (Exception ex)
```

Comment:
Can it throw anything else than JsonException actually? We should not catch knives.

Reply:
We aren't able to reference Newtonsoft.json in the shared project and therefore were not able to use a JsonException.

Reply:
ok